add console service test case for NameReg solidity contract

corbinbs opened the pull request:
Hi,

I added a console service test case to follow along with the https://github.com/ethereum/pyethapp/wiki/The_Console#creating-contracts section that I'd referenced in https://github.com/ethereum/pyethapp/pull/123   This may help out with some of the new issues like https://github.com/ethereum/pyethapp/issues/127   

The wiki uses `tx = namereg.register('alice')` but when I run that with the current develop branch, 
it ends up being reverted (may be like #127 is describing?)   I found that if I adjusted that to 
`namereg.register('alice', startgas=90000, gasprice=50 * 10**9)` it would run OK and `namereg.resolve(eth.coinbase)` would return the proper value.   

I'm not sure if different defaults are needed here https://github.com/ethereum/pyethapp/blob/develop/pyethapp/console_service.py#L194 or if the https://github.com/ethereum/pyethapp/wiki/The_Console wiki needs to be edited to include the startgas and gasprice parameters.

I'd like to help continue building up console service test cases also as I'm using it quite a bit now.   Let me know if I can extend anything with this to help out with #126 and #127 

Thanks,

Brian

corbinbs committed:
add console service test case for NameReg solidity contract

konradkonrad commented:
Thanks @corbinbs !
For me, this runs in an port issue because the jsonrpc port is contested. For testing the console service, you can remove the `JSONRPCServer` from https://github.com/corbinbs/pyethapp/blob/5a7697f1bbf28aef80f1c118a018e6742d9377ac/pyethapp/tests/test_console_service.py#L91, which should resolve the issue. ~~Can you fix that?~~ see below!

konradkonrad commented:
Also, we will need to add solidity to `.travis.yml`: https://travis-ci.org/ethereum/pyethapp/builds/129875926#L1250 -- I look into that!

konradkonrad committed:
Remove JSONRPCServer from console service test

It leads to port allocation errors and is not necessary in the scope
of the tests.

konradkonrad committed:
Add solidiy compiler to travis environment

konradkonrad commented:
please see https://github.com/corbinbs/pyethapp/pull/1

corbinbs committed:
Merge pull request #1 from ethereum/solc-on-travis-wip

Add solc and remove JSONRPCServer

corbinbs commented:
thanks again for the feedback and assist on this one, @konradkonrad 
Looks like the travis-ci build just passed and coverage bumped up 1.5%!
